src/impl_float_maths.rs

```diff
@@ -0,0 +1,162 @@
+        pub fn $id(&self) -> Array<A, D> {
+            self.mapv(A::$id)
+        })+
+    };
+}
+
+macro_rules! binary_op {
+    ($($(#[$meta:meta])* fn $id:ident($ty:ty))+) => {
+        $($(#[$meta])*
+        #[must_use = "method returns a new array and does not mutate the original value"]
+        pub fn $id(&self, rhs: $ty) -> Array<A, D> {
+            self.mapv(|v| A::$id(v, rhs))
+        })+
+    };
+}
+
+/// # Element-wise methods for Float Array
+///
+/// Element-wise math functions for any array type that contains float number.
+impl<A, S, D> ArrayBase<S, D>
+where
+    A: Float,
+    S: Data<Elem = A>,
+    D: Dimension,
+{
+    boolean_op! {
+        /// If the number is `NaN` (not a number), then `true` is returned for each element.
+        fn is_nan
+        /// Return `true` if any element is `NaN` (not a number).
+        fn is_any_nan -> is_nan
+
+        /// If the number is infinity, then `true` is returned for each element.
+        fn is_infinite
+        /// Return `true` if any element is infinity.
+        fn is_any_infinite -> is_infinite
+    }
+    unary_op! {
+        /// The largest integer less than or equal to each element.
+        fn floor
+
+        /// The smallest integer less than or equal to each element.
+        fn ceil
+
+        /// The nearest integer of each element.
+        fn round
+
+        /// The integer part of each element.
+        fn trunc
+
+        /// The fractional part of each element.
+        fn fract
+
+        /// Absolute of each element.
+        fn abs
+
+        /// Sign number of each element.
+        ///
+        /// + `1.0` for all positive numbers.
+        /// + `-1.0` for all negative numbers.
+        /// + `NaN` for all `NaN` (not a number).
+        fn signum
+
+        /// The reciprocal (inverse) of each element, `1/x`.
+        fn recip
+
+        /// Square root of each element.
+        fn sqrt
+
+        /// `e^x` of each element (exponential function).
+        fn exp
+
+        /// `2^x` of each element.
+        fn exp2
+
+        /// Natural logarithm of each element.
+        fn ln
+
+        /// Base 2 logarithm of each element.
+        fn log2
+
+        /// Base 10 logarithm of each element.
+        fn log10
+
+        /// Cubic root of each element.
+        fn cbrt
+
+        /// Sine of each element (in radians).
+        fn sin
+
+        /// Cosine of each element (in radians).
+        fn cos
+
+        /// Tangent of each element (in radians).
+        fn tan
+
+        /// Converts radians to degrees for each element.
+        fn to_degrees
+
+        /// Converts degrees to radians for each element.
+        fn to_radians
+    }
+    binary_op! {
+        /// Integer power of each element.
+        ///
+        /// This function is generally faster than using float power.
+        fn powi(i32)
+
+        /// Float power of each element.
+        fn powf(A)
```

Comment:
Food for thought: This method is "powf(array, scalar)". We need a separate name for `powf(array, array)`. What should the name of that method be?  I don't mean that we should have that in this PR, but sometime down the line.